lib/Barriers/PreDefined.pm

```diff
@@ -113,49 +125,93 @@ The number of the display decimal
 
 The method for the barrier calculation, method_1 or method_2
 
-=head2 config_param
+=head2 contract_barrier_levels
 
-A configuration parameter contains the configuration of each contract type
+A set of barrier level that intended to obtain for a contract type
 
 =cut
 
-my $config_param = LoadFile(File::ShareDir::dist_file('Barriers::PreDefined', 'contract_type_config.yml'));
+has [qw(
+       config_file
+       contract_type
+       duration
+       central_spot
+       display_decimal
+       method
```

Comment:
No need of `config_file`, as we'll transfer the `config` hashref.

All other parameters are transferred as arguments to the method, no need to have them as attributes.